src/containers/explore/EntityNeighborsTable.js

```diff
@@ -0,0 +1,177 @@
+   */
+
+  const dispatch = useDispatch();
+  const [tableData, setTableData] = useState([]);
+  const [tablePage, setTablePage] = useState(0);
+  const [neighborsIndex, setNeighborsIndex] = useState(1);
+  const fetchEntitySetDataRS :?RequestState = useRequestState([DATA, FETCH_ENTITY_SET_DATA, dataSetId]);
+
+  const neighborToAssociationEKIDs :Map = useMemo(() => (
+    Map().withMutations((mutableMap) => {
+      if (neighbors) {
+        neighbors
+          .slice(neighborsIndex, neighborsIndex + MAX_HITS_10)
+          .forEach((neighbor :Map) => mutableMap
+            .set(get(neighbor, 'neighborId'), get(neighbor, 'associationId')));
+      }
+    })
+  ), [neighbors, neighborsIndex]);
+
+  const totalNeighbors :number = useMemo(() => (
+    neighbors ? neighbors.count() : 0
+  ), [neighbors]);
+
+  // OPTIMIZE: no need to compute this on every render
+  const associationPropertyTypes :PropertyType[] = useEntityTypePropertyTypes(associationDataSet.get('entityTypeId'));
+  const neighborPropertyTypes :PropertyType[] = useEntityTypePropertyTypes(dataSet.get('entityTypeId'));
+
+  // OPTIMIZE: no need to compute this on every render
+  const tableHeaders = [];
+  associationPropertyTypes.forEach((propertyType) => tableHeaders.push({
+    key: `${propertyType.type.toString()}_edge`,
+    label: `${propertyType.title} (${propertyType.type.toString()}) (Edge)`,
+    sortable: false,
+    cellStyle: { 'background-color': PURPLE.P00 }
+  }));
+  neighborPropertyTypes.forEach((propertyType) => tableHeaders.push({
+    key: propertyType.type.toString(),
+    label: `${propertyType.title} (${propertyType.type.toString()})`,
+    sortable: false,
+  }));
+
+  // OPTIMIZE: no need to compute this on every render
+  const dataSetData :Map = useSelector(selectOrgEntitySetData(dataSetId, neighborToAssociationEKIDs.keySeq()));
+  const associationData :Map = useSelector(
+    selectOrgEntitySetData(associationDataSetId, neighborToAssociationEKIDs.valueSeq())
+  );
+
+  useEffect(() => {
+    if (fetchEntitySetDataRS === RequestStates.SUCCESS) {
+      const newTableData = neighborToAssociationEKIDs.entrySeq()
+        .map(([neighborEntityKeyId :UUID, associationEntityKeyId :UUID], index :number) => {
+          // 'id' is used as the "key" prop in the table component, so it needs to be unique
+          let entityData = dataSetData.get(neighborEntityKeyId).set('id', `${neighborEntityKeyId}-${index}`);
+          associationData.get(associationEntityKeyId).forEach((value, key) => {
+            entityData = entityData.set(`${key}_edge`, value);
+          });
+          return entityData;
+        }).toJS(); // TODO: avoid .toJS()
+      setTableData(newTableData);
+    }
+  // NOTE: leaving out "dataSetData" from depedency array because it tends to cause infinite renders
+  // TODO: figure out how to avoid the infinite renders when "dataSetData" is passed
```

Comment:
This is because dataSetData is computed from selected values. you should instead be moving the computation inside the effect, rather than in the selector. I think this is an issue throughout. 

Whatever computation the selectors are doing should be broken out into external functions to be used inside these effects. We can also use reselect, which memoizes selectors.

Reply:
looking into this now.

Reply:
what computation are you referring to that needs to be made in the effect?

Reply:
`selectOrgEntitySetData` creates a new Map every time because it transforms the values kept in the store into some other format. This transformation needs to either happen here in the effects, or the selector should be memoized.

@UnsungHero97 has attempted to work around this by placing object hashes as dependencies. This not only requires additional calculations, but also incurs a different type of warning where the listed dependency is not used.

Either relocating the transform into the effect or memoizing the selector will resolve both of these issues.